ivy/__init__.py

```diff
@@ -978,7 +978,7 @@ def del_global_attr(attr_name):
     delattr(globals_vars, attr_name)
 
 
-backend = ""
+backend = ivy.utils.backend.handler.implicit_backend
```

Comment:
I don't think we should be doing this, because as a result of this change `ivy.backend` returns `numpy` and so does `ivy.current_backend_str()`. However, if you haven't set a backend and still pass a `torch.Tensor` to an `ivy` function, it would defer to the `torch` backend implementation, which isn't the same as when the `numpy` backend is actually set. 
We should ideally retain the current behaviour when a backend isn't set, what do you think?

Reply:
we can still differentiate between implicit and numpy backend as I mentioned above by `len(ivy.backend_stack)`, it's demonstrated in the change in `sub_backend_handler.py`, aside from that everything is working as expected
![image](https://github.com/unifyai/ivy/assets/32404268/8ee5e0be-2952-46d2-901d-bf43fb2e39fd)

Reply:
I'm not sure if it will be used as an indicator through, `ivy.backend` being `numpy` would end up being assumed as though the backend set is `numpy`. I think the difference in `ivy.backend` with and without setting the backend is a much clearer indicator that `numpy` would only be used if the input passed to the function doesn't belong to any specific backend (as `current_backend()` returns `numpy` in this case, but the appropriate backend if the input belongs to another backend). I think the previous setup of `numpy` being the "default" backend rather than the "implicit" one is less ambiguous about the usability of `ivy` without setting a backend

Reply:
.